source/fuzz/fuzzer_pass.cpp

```diff
@@ -183,9 +186,23 @@ void FuzzerPass::ForEachInstructionWithInstructionDescriptor(
 }
 
 void FuzzerPass::ApplyTransformation(const Transformation& transformation) {
-  assert(transformation.IsApplicable(GetIRContext(),
-                                     *GetTransformationContext()) &&
-         "Transformation should be applicable by construction.");
+  if (ignore_inapplicable_transformations_) {
+    // If an applicable-by-construction transformation turns out to be
+    // inapplicable, this is a bug in the fuzzer. However, when deploying the
+    // fuzzer at scale for finding bugs in SPIR-V processing tools it is
+    // desirable to silently ignore such bugs. This code path caters for that
+    // scenario.
+    if (!transformation.IsApplicable(GetIRContext(),
+                                     *GetTransformationContext())) {
+      return;
+    }
+  } else {
+    // This code path caters for debugging bugs in the fuzzer, where an
+    // applicable-by-construction transformation turns out to be inapplicable.
+    assert(transformation.IsApplicable(GetIRContext(),
```

Comment:
A separate note. It might be better to have some sort of runtime assertions if we decide to run fuzzer on release mode.

Reply:
I'm not sure what you mean - do you mean so that we would do this check even in release mode?

Reply:
Yes. I think it's good to have an assertion here even in release mode since applying an inapplicable transformation might break things a lot. What do you think?

Reply:
I think I'm OK with there being no assertion in release mode. You can build with Release+Asserts to get the assertions in an otherwise optimized build. When we hopefully get the fuzzer to the point that these assertions really don't fail, I think it's useful to have the option of fuzzing at speed on the assumption that they won't fail.

Reply:
Sounds good to me.